cpp/tensorrt_llm/thop/allreduceOp.cpp

```diff
@@ -310,6 +313,120 @@ class AllreduceOp
         return {};
     }
 
+    std::vector<torch::Tensor> runLowPrecisionAllReduce(torch::Tensor const& input,
+        torch::optional<torch::Tensor> const& residual, torch::optional<torch::Tensor> const& norm_weight,
+        torch::optional<torch::Tensor> const& scale, torch::optional<torch::Tensor> const& bias,
+        torch::optional<torch::Tensor> workspace, AllReduceStrategyType strategy) noexcept
+    {
+#ifdef ENABLE_FP8
+        auto stream = at::cuda::getCurrentCUDAStream(input.get_device());
+        int size = input.numel();
+        int hidden_size = input.size(-1);
+
+        auto const tp_size = mGroup.size();
+        auto const cur_rank = COMM_SESSION.getRank();
+        int tp_rank = 0;
+
+        for (auto const& currentRank : mGroup)
+        {
+            if (cur_rank == currentRank)
+                break;
+            ++tp_rank;
+        }
+
+        int bytes_per_element = input.element_size();
+
+        int token_num = size / input.size(-1);
```

Comment:
```suggestion
        int token_num = size / hidden_size;
```
Avoid calling Tensor.size() over and over again. It's not free.

Reply:
Thank you, I have made the modifications according to your comment.